### ISSUE TITLE
ICO: java.lang.IllegalArgumentException: Pixel stride times width must be less than or equal to the scanline stride

### ISSUE THREAD

**Issue by xinyingho** [haraldk/TwelveMonkeys#150]
I got a set of ico files that raises an illegal exception while reading them. The code is quite simple:
```java
ImageInputStream stream = ImageIO.createImageInputStream(Files.newInputStream(
                                    path, StandardOpenOption.READ
));
ImageReader reader = ImageIO.getImageReaders(stream).next();
reader.setInput(stream);
BufferedImage image = reader.read(0, null);
```
Here's one of the problematic ico files:
![18w](https://cloud.githubusercontent.com/assets/9655252/8568536/b7730802-256e-11e5-812c-93ac9b7bae3c.png)

Windows Explorer and other image editing applications can display them all right. So I suppose that the BMP library is not lenient enough towards malformed ico files.

**Comment by haraldk** [comment]
Hi,

Thanks for reporting! I can reproduce the problem, will look into it as soon as possible.


-- 
Harald K

**Comment by haraldk** [comment]
Fixed in master (d1cb941f06abdbf9f1c725649bd6bb5eb444fb5c).

-- 
Harald K
